sklearn/cluster/k_means_.py

```diff
@@ -1282,15 +1290,20 @@ def fit(self, X, y=None):
             Coordinates of the data points to cluster
         """
         random_state = check_random_state(self.random_state)
-        X = check_array(X, accept_sparse="csr", order='C', dtype=np.float64)
+        # to handle sparse data which only works as float64 at the moment
```

Comment:
If this is not trivial to do, I am ok with keeping the sparse case for a second PR.